### DISCUSSION TITLE
Help to setup Twitter notifications

### DISCUSSION THREAD

**Discussion by lozbinks** [jef/streetmerchant#2047] [Help]
Does anyone know how to set the twitter notifications up or is able to link to a guide?

I have added the required tokens and keys for twitter to my dotenv however, I feel something else is missing as its not sending the tweets.

Would @bryanberger be able to help as I notice from a previous post that you may be the best person to ask about this area.

Many thanks

**Comment by bryanberger** [comment]
@lozbinks are you getting any console output? It should be as simple as setting these `vars` https://jef.codes/streetmerchant/reference/notification/#twitter

you can test by running `npm run test:notification`

**Reply by lozbinks** [reply]
Thanks very much for the reply @bryanberger, yes everything appears to be running as it should in the console. I've run npm run test:notification however it only sends a desktop notification.

Here is where I've placed my Twitter keys and tokens in my dotenv as per the above link 

![twitter keys in dotenv](https://user-images.githubusercontent.com/78455265/109438787-4b421c80-7a23-11eb-8396-7d3c17538d51.PNG)

I've also created the keys with read and write permissions from within the Twitter developer portal.

For my TWITTER_ACCESS_TOKEN_KEY I've used the key from "access token" 
For my TWITTER_CONSUMER_KEY I've used the key from "API Key" which I understand is the consumer key
For my TWITTER_CONSUMER_SECRET I've used the "API secret"

**Comment by lozbinks** [comment]
@bryanberger Ah I've worked it out now! 

My dotenv didn't have the TWITTER_ACCESS_TOKEN_SECRET line for some reason so I hadn't added that (my bad). I knew it would be something really simple that I'd most likely just overlooked.

Thanks again for your assistance!
